### PR TITLE
Add asprintf implementation

### LINKED ISSUE #15: Add asprintf/vasprintf function
This is needed by util-linux

https://man7.org/linux/man-pages/man3/asprintf.3.html
https://www.freebsd.org/cgi/man.cgi?query=asprintf&apropos=0&sektion=3&manpath=FreeBSD+6.2-RELEASE&format=html

It can be created by wrapping vsnprintf, eg something like this but with error checking:

```c
int size = vsnprintf(NULL, 0, fmt, args);
char* str = malloc(size+1);
vsnprintf(str, size+1, fmt, args);
```

### PR DESCRIPTION
**Pull request by naveen-ram**
Useful for minisign and future packages

Fixes #15

### CODE REVIEW

File: util/build.json [hunk 1]
```diff
@@ -18,7 +19,8 @@
             "getopt.h",
             "pty.h",
             "execinfo.h",
-            ["file.h", "sys/file.h"]
+            ["file.h", "sys/file.h"],
+            "asprint.h"
```
Comment (kadler) [inline util/build.json:23]:
```suggestion
            ["stdio.h", "stdio.h"]
```

asprintf and vasprintf are expected to be in [stdio.h](https://man7.org/linux/man-pages/man3/asprintf.3.html), so this header should be named the same.

Please copy what was done for flock in https://github.com/IBM/portlibfori/commit/a0a6b52e637ade3da99247356dd63f7bfffeade3#diff-630f36c8ca89eabc3e205055d1cfc267. You'll need to add the `#include_next` and we should add the `__asm__` bit too, to rename the symbols with the libutil_ prefix.

**Review by kadler — CHANGES_REQUESTED** [review]

**Review by ThePrez — APPROVED** [review]

File: util/asprintf.c [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#include <stdlib.h>
+#include <stdio.h>
+
+int libutil_vasprintf(char **ret, const char *format, va_list args)
+{
+    *ret = NULL;
+
+    va_list args_copy;
+    va_copy(args_copy, args)
+    int size = vsnprintf(NULL, 0, format, args);
```
Comment (kadler) [inline util/asprintf.c:11]:
```suggestion
    int size = vsnprintf(NULL, 0, format, args_copy);
```

File: util/asprintf.c [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    char* str = malloc(size+1);
+    
+    if (str == NULL) return -1;
+
+    size = vsnprintf(str, (size_t) size + 1, format, args);
+    
+    if (size == -1)
+    {
+        free(str);
+        return -1
```
Comment (kadler) [inline util/asprintf.c:25]:
```suggestion
        return -1;
```

File: util/stdio.h [hunk 4]
```diff
@@ -0,0 +1,21 @@
+  
+#ifndef LIBUTIL_STDIO_H
+#define LIBUTIL_STDIO_H
+
+/* Include the real sys/file.h. We don't want to recreate all the stuff */
+/* it contains - just override flock */
```
Comment (kadler) [inline util/stdio.h:6]: Looks like this comment needs to be adjusted since it still references sys/file.h and flock.

File: util/libutil.exp [hunk 5]
```diff
@@ -9,3 +9,5 @@ backtrace
 backtrace_symbols
 backtrace_symbols_fd
 libutil_flock
+asprintf
+vasprintf
```
Comment (kadler) [inline util/libutil.exp:13]: These need the libutil_ prefix to match the function definitions.

**Review by kadler — CHANGES_REQUESTED** [review]
Looking pretty good, just a few small things to clean up.

**Review by kadler — APPROVED** [review]
LGTM

**Pull request merged**